projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -121,8 +122,8 @@ def initialize_job(self, context: JobContext) -> None:
 
         handler = logging.StreamHandler(sys.stderr)
         handler.setFormatter(formatter)
-        handler.addFilter(metadata_filter)
```

Comment:
why is the order changed?

Reply:
Filters execute in sequence. If you add the job name after you filtered the metadata, filtering the job name in the metadata will not work. You'll be stuck with it til the end of the execution.

Reply:
This is currently a problem with `vdk_step_name` and `vdk_step_type`. I opened a separate ticket for it https://github.com/vmware/versatile-data-kit/issues/2875